qiita_pet/handlers/study_handlers/listing_handlers.py

```diff
@@ -42,14 +43,82 @@ def _get_shared_links_for_study(study):
     return ", ".join(shared)
 
 
-def _build_study_info(user, results=None):
-    """builds list of dicts for studies table, with all html formatted"""
+def _build_single_study_info(study, info):
+    # Clean up and add to the study info for HTML purposes
+    PI = StudyPerson(info['principal_investigator_id'])
+    status = study.status
+    if info['pmid'] is not None:
+        info['pmid'] = ", ".join([pubmed_linkifier([p])
+                                  for p in info['pmid']])
+    else:
+        info['pmid'] = ""
+    if info["number_samples_collected"] is None:
+        info["number_samples_collected"] = 0
+    info["shared"] = _get_shared_links_for_study(study)
+    info["num_raw_data"] = len(study.raw_data())
+    info["status"] = status
+    info["study_id"] = study.id
+    info["pi"] = study_person_linkifier((PI.email, PI.name))
+    del info["principal_investigator_id"]
+    del info["email"]
+    return info
+
+
+def _build_proc_data_info(study, study_proc, proc_samples):
```

Comment:
IMOO, this function should follow the structure of the previous one. It should be `_build_single_proc_data_info` and then you call it in a list comprehension from `_build_single_study_info`. You will have then a function for a single object, and the list comprehension will be cheaper than the appends.